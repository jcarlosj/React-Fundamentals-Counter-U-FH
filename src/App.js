import React from 'react';

/** Functional Component */
const App = ({ name }) => {

    const data = {
        name,
        title: `Hola ${ name } desde React!`,
        message: 'Usamos un Fragment para agrupar elementos HTML y poder retornarlos como uno solo'
    };

    console.log( name );

    return (
        <> 
            <h1>{ data.title }</h1>
            <p>{ data.message }</p> 
            <pre>{ JSON.stringify( data, null, 4 ) }</pre>
        </>
    );
}
 
export default App;