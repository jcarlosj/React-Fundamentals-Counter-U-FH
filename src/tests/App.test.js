// import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import App from '../App';

describe( '<App />', () => {

//     test( 'it should display the message "Hola, Eva Sofia Janeth! (Forma 1)', () => {
    
//         const 
//             name = 'Eva Sofia Janeth',
//             title = 'Hola, Eva Sofia Janeth!',
//             { getByText } = render( <App name={ name } /> );        //  Monta el Componente de React & Extrae la propiedad del Componente

//         expect( getByText( title ) ).toBeInTheDocument();           //  Usa la propiedad del Componente

//     });

//     test( 'it should display the message "Hola, Eva Sofia Janeth! (Forma 2)', () => {
    
//         const 
//             name = 'Eva Sofia Janeth',
//             title = 'Hola, Eva Sofia Janeth!';
            
//         render( <App name={ name } /> );    //  Monta el Componente de React

//         expect( screen.getByText( title ) ).toBeInTheDocument();    //  Usa screen para hacer uso de la propiedad del componente

//     });

    describe( 'Enzyme:', () => {
        
        test( 'it should display the message "Hola, Eva Sofia Janeth!', () => {
            
            const 
                name = 'Eva Sofia Janeth', 
                wrapper = shallow( <App name={ name } /> );

            // console.log( wrapper.debug() );

            expect( wrapper ).toMatchSnapshot();    //  Verifica que un valor coincida con la instantánea (Snapshop) más reciente
            
        });

    }); 

});

