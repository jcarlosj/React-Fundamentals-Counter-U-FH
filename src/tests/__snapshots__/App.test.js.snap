// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Enzyme: it should display the message "Hola, Eva Sofia Janeth! 1`] = `
<Fragment>
  <h1>
    Hola, Eva Sofia Janeth!
  </h1>
  <p>
    React te da la bienvenida
  </p>
  <pre>
    {
    "name": "Eva Sofia Janeth",
    "title": "Hola, Eva Sofia Janeth!",
    "message": "React te da la bienvenida"
}
  </pre>
</Fragment>
`;
